test-utils/src/monerod.rs

```diff
@@ -34,7 +37,10 @@ pub async fn monerod<T: AsRef<OsStr>>(flags: impl IntoIterator<Item = T>) -> Spa
     let p2p_port = get_available_port(&[rpc_port]);
     let zmq_port = get_available_port(&[rpc_port, p2p_port]);
 
-    // TODO: set a random DB location
+    let db_location = current_dir()
+        .unwrap()
+        .join(format!("monerod_data_{}", rand::random::<u64>()));
```

Comment:
I couldn't get `tempfile` to work - it wasn't using `/tmp` and wasn't dlelting the dir when finished like to docs say so I just went with stds `temp_dir`

Reply:
`tempfile::tempdir()` returns the same directory as `temp_dir` since it uses it internally. 

It's just a test though so `temp_dir` is ok.

Reply:
thats what the docs say but it was putting the dir in `monero-p2p`

Reply:
Hmm I doubt `tempfile` is incorrect, it just calls [`temp_dir`](https://docs.rs/tempfile/latest/src/tempfile/lib.rs.html#597). Maybe some CI env problem.

Reply:
This issue was because of 
```rust
 .arg(format!(r"--data-dir='{}'", db_location.display()))
 ```
 this was making monerod put the file in the same folder that we are currently in